### DISCUSSION TITLE
Configure collider as a child entity/component

### DISCUSSION THREAD

**Discussion by wentao** [avianphysics/avian#321] [General]
Here is the use case I'm facing: I have a blender exported glb scene where it contains a rendering object and collider object with different names. Let's call them `view` and `collider`. The `collider` object is much simpler but wraps the `view` object very well.

After loading the scene into bevy, I can use the following code to hide the `collider` from rendering
```
commands
        .spawn((
            HookedSceneBundle {
                scene: SceneBundle {
                    transform: ...
                    scene: ...
                    ..default()
                },
                hook: SceneHook::new(|entity, cmds| {
                    let name = entity.get::<Name>().map(|t| t.as_str());
                    match name {
                        Some("collider") => cmds
                            .insert(Visibility::Hidden)
                            .insert(TagCollider),
                        _ => cmds,
                    };
                }),
            },
            RigidBody::Kinematic,
```

The `TagCollider` component will be processed async in another system where we can generate the `Collider` from the mesh

```
let collider = Collider::trimesh_from_mesh(meshes.get(mesh.id()).unwrap()).unwrap();
commands.entity(entity).remove::<TagCollider>().insert(collider);
```

This works fine and I can detect collision events between the colliders correct. However, the `Collider` is attached to a descendant entity of the root entity I added during the scene loading phase.

Is there anyway to configure collision event to point at `RigidBody` entity (the root), rather than the `Collider` entity? I used `bevy_rapid3d` previously and the collision events generated there always points `RigidBody` entity instead of `Collider` entity.

**Comment by Jondolf** [comment]
Child colliders have a `ColliderParent` component that points to the `Entity` of the root `RigidBody`. That should work as a workaround for now.

It would be straightforward to change collision events to always point to the rigid body entity, and I might change the default to that... but pointing to the `Collider` entity can also have use cases. Like if you had a sensor collider for ground detection as the child of a `Player` entity, you might want to get collisions specifically for the collider to know that you hit the ground and not something else.

One option would be to make collisions store both the rigid body entity and the collider entity. This would make both use cases viable.

**Reply by wentao** [reply]
Thank you for quick response!

`ColliderParent` works pretty well for my use case. Having collisions store both regid body entity and collider entity sounds even better.
